pkg/registry/core/node/storage/storage.go

```diff
@@ -104,8 +103,8 @@ func NewStorage(optsGetter generic.RESTOptionsGetter, kubeletClientConfig client
 	proxyREST := &noderest.ProxyREST{Store: store, ProxyTransport: proxyTransport}
 
 	// Build a NodeGetter that looks up nodes using the REST handler
-	nodeGetter := client.NodeGetterFunc(func(nodeName string, options metav1.GetOptions) (*v1.Node, error) {
-		obj, err := nodeREST.Get(genericapirequest.NewContext(), nodeName, &options)
+	nodeGetter := client.NodeGetterFunc(func(nodeName string, options metav1.GetOptions, ctx context.Context) (*v1.Node, error) {
```

Comment:
Looking at usage, I don't think we want this at all.  Why wouldn't we just a loopback connection like we do for things like RBAC?

Reply:
agree. then we can remove explicit conversion like:[https://github.com/kubernetes/kubernetes/blob/e9a3aaa44b0655e0ca67be6d460a4ef0774e0648/pkg/registry/core/node/storage/storage.go#L118](https://github.com/kubernetes/kubernetes/blob/e9a3aaa44b0655e0ca67be6d460a4ef0774e0648/pkg/registry/core/node/storage/storage.go#L118)

Reply:
will put it in another issue/pull 👍🏻

Reply:
it'll take some extra cost though. but for now it looks okay b/c these stream storages are not frequent caller